torch/jit/_monkeytype_config.py

```diff
@@ -0,0 +1,115 @@
+
+_IS_MONKEYTYPE_INSTALLED = True
+try:
+    import monkeytype
+    from monkeytype import trace as monkeytype_trace
+    from monkeytype.db.base import CallTraceStore, CallTraceStoreLogger
+    from monkeytype.config import default_code_filter
+    from monkeytype.tracing import CallTrace, CodeFilter
+except ImportError:
+    _IS_MONKEYTYPE_INSTALLED = False
+    print("Warning: monkeytype is not installed. Please install monkeytype to enable Profile-Directed Typing in TorchScript")
+
+def get_qualified_name(func):
+    return func.__qualname__
+
+if _IS_MONKEYTYPE_INSTALLED:
+
+    class JitTypeTraceStoreLogger(CallTraceStoreLogger):
+        """A JitTypeCallTraceLogger that stores logged traces in a CallTraceStore."""
+        def __init__(self, store: CallTraceStore):
+            super().__init__(store)
+
+        def log(self, trace: CallTrace) -> None:
+            self.traces.append(trace)
+
+    class JitTypeTraceStore(CallTraceStore):
+        def __init__(self):
+            super().__init__()
+            # A dictionary keeping all collected CallTrace
+            # key is fully qualified name of called function
+            # value is list of all CallTrace
+            self.trace_records = defaultdict(list)
+
+        def add(self, traces: Iterable[CallTrace]):
+            for t in traces:
+                qualified_name = get_qualified_name(t.func)
+                self.trace_records[qualified_name].append(t)
+
+        def filter(
+            self,
+            qualified_name: str,
+            qualname_prefix: Optional[str] = None,
+            limit: int = 2000
+        ) -> List[CallTrace]:
+            return self.trace_records[qualified_name]
+
+        def analyze(self, qualified_name: str) -> Dict:
+            # Analyze the types for the given module
+            # and create a dictionary of all the types
+            # for arguments and return values this module can take
+            records = self.trace_records[qualified_name]
+            all_args = defaultdict(set)
+            for record in records:
+                for arg, arg_type in record.arg_types.items():
+                    all_args[arg].add(arg_type)
+                all_args["return_type"].add(record.return_type)
+            return all_args
+
+        def consolidate_types(self, qualified_name: str) -> Dict:
+            all_args = self.analyze(qualified_name)
+            # If there are more types for an argument,
+            # then consolidate the type to `Any` and replace the entry
+            # by type `Any`.
+            # This currently handles only cases with only one return value
+            # TODO: Consolidate types for multiple return values
+            for args, types in all_args.items():
+                if len(types) > 1:
+                    all_args[args] = {'Any'}
+                else:
+                    _element_string_type = types.pop()
+                    all_args[args] = {_element_string_type.__name__}
```

Comment:
Does this work for container types like `List[int]`?

Reply:
I have added support for container types in the new fix.